pytorch_lightning/trainer/training_loop.py

```diff
@@ -837,6 +834,14 @@ def backward(self, result, optimizer, opt_idx, *args, **kwargs):
                 result.closure_loss, optimizer, opt_idx, *args, **kwargs
             )
 
+        if not self.should_accumulate():
+            # unscale gradients only when accumulated
+            if self.trainer.amp_backend == AMPType.NATIVE:
+                self.trainer.scaler.unscale_(optimizer)
```

Comment:
@tchaton what's your thoughts on bringing this into the training loop? It was nice being hidden within the accelerator, however the scaler object is created in the trainer code anyway... It would be nice to hide the internals of this function `accelerator.unscale(optimizer)` in my opinion but I'm indifferent on the decision made here!

Reply:
I have moved it within AMP Native Plugin. Makes more sense to me.

Reply:
I was too slow, nice fix :)